context/auth.tsx

```diff
@@ -1,18 +1,53 @@
-import React from "react"
-import { loadFirebase } from "./firebase"
-import axios from "../util/axios"
+import React, { PropsWithChildren } from "react"
+import { loadFirebase, userType } from "./firebase"
+import axios, { AxiosResponse } from "../util/axios"
+import { ProfileSerializer } from "../types/backend"
 
-const AuthContext = React.createContext({})
+type tokenType = string | null
+type profileType = ProfileSerializer | null
+type loadingType = boolean
+type PropTypes = PropsWithChildren<{}>
 
-export const AuthProvider = ({ children }) => {
-    const [firebaseUser, setFirebaseUser] = React.useState(null)
-    const [token, setToken] = React.useState(null)
-    const [profile, setProfile] = React.useState(null)
-    const [loading, setLoading] = React.useState(true)
+const AuthContext = React.createContext<{
+    handleLogout: () => void
+    handleSignIn: () => void
+    setFirebaseUser: (user: userType) => void
+    setProfile: (profile: profileType) => void
+    setToken: (token: tokenType) => void
```

Comment:
Oh, I just noticed these are just state hook dispatch functions. E.g, the `setLoading` has the type `React.Dispatch<React.SetStateAction<boolean>>`.

Btw, we may just skip the default value in the auth context for now since we were using this `AuthProvider` in `_app.tsx`.

Reply:
I have changed the types